snakemake/satija/src/generate_cell_sets.py

```diff
@@ -0,0 +1,185 @@
+import networkx
+
+from constants import COLUMNS, CL_ROOT_ID
+from utils import (
+    load_cl_obo_graph,
+    init_cell_sets_tree,
+    dict_to_tree,
+    sort_paths_up_cell_ontology
+)
+
+
+# Generate a tree of cell sets representing
+# the clusters from the Leiden clustering
+# algorithm.
+def generate_leiden_cluster_cell_sets(df):
```

Comment:
```suggestion
def generate_leiden_cluster_cell_sets(df):
    '''
    Generate a tree of cell sets representing
    the clusters from the Leiden clustering
    algorithm.
    '''
```
Slight preference.

Would doctests be feasible and useful here? The end-to-end tests are something... but if there is mismatch against the fixtures, it can be hard to work backward from them to find the root of the problem.